tests/Unit/Google/GlobalSiteTagTest.php

```diff
@@ -233,4 +238,62 @@ public function test_enhanced_conversion_data_is_set_with_customer_address() {
 		$this->assertStringContainsString( $last_hash, $gtag );
 		$this->assertStringContainsString( $postcode, $gtag );
 	}
+
+	public function test_register_initializes_gtg_adapter_with_conversion_id() {
+		$conversion_action = [
+			'conversion_id'    => 'AW-123456789',
+			'conversion_label' => 'abcDEFghi',
+		];
+
+		// Provide return values for both calls in register(): ADS_CONVERSION_ACTION and ADS_GTG_ENABLED.
+		/** @var MockObject|OptionsInterface $options_mock */
+		$options_mock = $this->options;
+		$options_mock->method( 'get' )
+			->willReturnMap(
+				[
+					[ OptionsInterface::ADS_CONVERSION_ACTION, null, $conversion_action ],
+					[ OptionsInterface::ADS_GTG_ENABLED, null, true ],
+				]
+			);
+
+		$light_tag = new class($this->assets_handler, $this->gtag_js, $this->product_helper, $this->wc, $this->wp) extends GlobalSiteTag { // phpcs:disable PSR12.Classes.AnonClassDeclaration.SpaceAfterKeyword
+			protected function register_assets() {}
+			protected function product_data_hooks() {}
+		};
+		$light_tag->set_options_object( $this->options );
+
+		// We can't directly intercept the external Adapter static creation now. This test simply
+		// ensures register() completes without throwing when a valid conversion action exists
+		// and GTG is enabled.
+		$light_tag->register();
+		$this->assertTrue( true );
+	}
+
+	public function test_register_exits_early_without_conversion_action() {
+		/** @var MockObject|OptionsInterface $options_mock */
+		$options_mock = $this->options;
+		$options_mock->method( 'get' )
+			->willReturnMap(
+				[
+					[ OptionsInterface::ADS_CONVERSION_ACTION, null, null ],
+					[ OptionsInterface::ADS_GTG_ENABLED, null, true ],
+				]
+			);
+
+		$instrumented = new class($this->assets_handler, $this->gtag_js, $this->product_helper, $this->wc, $this->wp) extends GlobalSiteTag { // phpcs:disable PSR12.Classes.AnonClassDeclaration.SpaceAfterKeyword
+			public $register_assets_called    = false;
+			public $product_data_hooks_called = false;
+			protected function register_assets() {
+				$this->register_assets_called = true; }
+			protected function product_data_hooks() {
+				$this->product_data_hooks_called = true; }
+		};
+		$instrumented->set_options_object( $this->options );
+
+		$instrumented->register();
+
+		$this->assertFalse( $instrumented->register_assets_called, 'register_assets() should not run when conversion action missing.' );
+		$this->assertFalse( $instrumented->product_data_hooks_called, 'product_data_hooks() should not run when conversion action missing.' );
+		$this->assertFalse( has_filter( 'woocommerce_loop_add_to_cart_link' ), 'Product data filter unexpectedly added.' );
```

Comment:
Let's add these check in `test_register_initializes_gtg_adapter_with_conversion_id` test that verify it return `true`